File: Core/Entities/Servant.cs
using RR.Core.Entities.Base;

namespace RR.Core.Entities;

public class Servant:BaseEntity
{
    public int Id { get; set; }

    public int AccountId { get; set; }

    //navegaçao via lazy loading
    public virtual Account Account { get; set; }
    public virtual ICollection<StudentAdvisor> AdvisedStudents { get; set; } = new List<StudentAdvisor>();
    public virtual ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
} 

File: RR.Application/Client/src/app/domain/models/servant.ts
import { BaseModel } from "./base-model";
import { Account } from "./account";

/**
 * Interface Servant - representa um servidor no sistema
 * Baseado na entidade C#: RR.Core.Entities.Servant
 */
export interface Servant extends BaseModel {
  Id: number;
  AccountId: number;
  Account?: Account | null; // Lazy loading - pode vir nulo

  // Herdado de BaseModel/BaseEntity
  CreatedAt?: Date | string;
  UpdatedAt?: Date | string;
  IsActive?: boolean;
}

/**
 * DTO simplificado para listagens e operações básicas
 * Usado nos services
 */
export interface ServantDto {
  Id: number;
  AccountId: number;
  Account?: Account | null;
  account?: Account | null; // case insensitive fallback
  IsActive?: boolean;
  isActive?: boolean; // case insensitive fallback
  CreatedAt?: Date | string;
  UpdatedAt?: Date | string;
}

